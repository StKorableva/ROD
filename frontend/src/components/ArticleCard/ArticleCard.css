.articleCard{
    /* Цвет для наглядности, пождразумевается использование background-img из компонента */
    background-color: rgb(250, 185, 87);
    /* Цвет для наглядности, пождразумевается использование background-img из компонента */
    width: 335px;
    height: 418px;
    position: relative;
}
.articleCard__contentBox{
    width: 319px;
    height: 119px;
    position: absolute;
    bottom: 12px;
    left:8px;
    background-color: rgba(22, 21, 21, 0.7);
}
.articleCard__contentBox p{
    padding-left: 12px;
    text-align: left;
    color: #BFBFBF;
}
.articleCard__contentBox__section{
    font-style: Bold;
    font-size: 12px;
    margin-top:8px;}
.articleCard__contentBox__heading{
    font-style: Bold;
    font-size: 24px;
    margin-top:15px;
}
.articleCard__contentBox__description{
    font-style: Regular;
    font-size: 12px;
}