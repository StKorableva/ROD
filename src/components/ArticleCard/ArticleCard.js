import './ArticleCard.css';

function ArticleCard(props) {
	const { imageUrl, openArticlePage, section, heading, description } = props;
	return (
		<div className="articleCard" style={{ backgroundImage: `url(${imageUrl})` }} onClick={openArticlePage}>
			<div className="articleCard__contentBox">
				<p className="articleCard__contentBox__section">{section}</p>
				<p className="articleCard__contentBox__heading">{heading}</p>
				<p className="articleCard__contentBox__description">{description}</p>
			</div>
		</div>
	);
}

export default ArticleCard;
