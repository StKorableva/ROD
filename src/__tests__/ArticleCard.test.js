import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ArticleCard from '../components/ArticleCard/ArticleCard';

describe('base test', () => {
	const section = 'Test section',
		heading = 'Test heading',
		description = 'Test description';
	const component = shallow(<ArticleCard section={section} heading={heading} description={description}></ArticleCard>);

	it('render test', () => {
		expect(component).toHaveLength(1);
	});

	it('snapshot test', () => {
		expect(toJSON(component)).toMatchSnapshot();
	});

	it('in the document', () => {
		expect(component.text()).toMatch(section);
		expect(component.text()).toMatch(heading);
		expect(component.text()).toMatch(description);
	});
});
