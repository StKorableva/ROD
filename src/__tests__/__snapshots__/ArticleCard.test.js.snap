// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base test snapshot test 1`] = `
<div
  className="articleCard"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
>
  <div
    className="articleCard__contentBox"
  >
    <p
      className="articleCard__contentBox__section"
    >
      Test section
    </p>
    <p
      className="articleCard__contentBox__heading"
    >
      Test heading
    </p>
    <p
      className="articleCard__contentBox__description"
    >
      Test description
    </p>
  </div>
</div>
`;
